watchgod/watcher.py

```diff
@@ -82,12 +82,10 @@ def should_watch_file(self, entry):
 
 class RegExpWatcher(AllWatcher):
     def __init__(self, root_path, re_files=None, re_dirs=None):
-        self.re_files = re.compile(re_files) if re_files is not None else re_files
-        self.re_dirs = re.compile(re_dirs) if re_dirs is not None else re_dirs
+        if re_files is not None:
+            self.re_files = re.compile(re_files)
+            self.should_watch_file = lambda entry: self.re_files.match(entry.path)
```

Comment:
I would like to avoid using an if statement in the method because of its extensive utilization. On the other hand, I admit that this pull request violates any future inheritance of RegExpWatcher. Shall I work your proposal?

Reply:
Do it the simple and obvious way (eg. with a method on the class and an if) for now.

Long term I'm intending to move to rust's notify package for the heavy lifting which will mean these methods are no longer called quite so much.

Reply:
Is this OK?
~~~python
def should_watch_file(self, entry):
        if self.re_files is not None:
            return self.re_files.match(entry.path)

        return super().should_watch_file(entry)
~~~

Reply:
looks good, in cases like this i normally use an `else` for clarity but it's not essential.